### ISSUE TITLE
[Request] "Collective Sum of Bests" would describe "theoretical perfect run"

### ISSUE THREAD

**Issue by jhobz** [glacials/splits-io#190]
A collective sum of bests feature that takes the best segment time out of all runs submitted (and their run history) for each segment would be kind of cool to describe a theoretical "perfect run" or TAS time. This would obviously be aided greatly by the implementation of "alternate" (communal) segment names by LiveSplit (see https://github.com/LiveSplit/LiveSplit/issues/115).

**Comment by glacials** [comment]
Very interesting, thanks for linking that LiveSplit issue. I do think that split IDs would help a lot here, as it's a lot of guesswork to figure out if one set of splits looks like another.

One of the ideas I've been toying around with has been a more branded concept of routes, where you can upload blank splits and advertise them not as splits for a category but as a route for a category. This issue opening up made me think about it a bunch more and I just brain dumped all over #67.

I also see a routes feature as a prerequisite to theoretical perfect runs, as parsing all splits of all runs in a category to discover split IDs before knowing which ones to pull splits from is too much work to do on demand -- we'd instead need an efficient way to find all runs of a given route, which we could do by just figuring it out at upload-time and remembering that that run belonged to that route.

**Comment by glacials** [comment]
Oops I put this on the wrong thread three days ago, moving here:

> Even with identical splits, because of differences between runners there are no guarantees that the global sum of bests is actually attainable. For example in Sunshine, everyone splits when you get a shine. But the "shine get" cutscene is about 4 seconds long; some people split when it starts and some people split when it ends.

> If you split when it starts, your first segment is shorter by 4 seconds than that of a person who splits when it ends. And if the game ended with a shine get cutscene (it doesn't), that person's last split would be shorter than yours by 4 seconds. So now the global sum of bests is 8 seconds faster than what's actually proven to be achievable.

> I don't think this is a particularly huge deal, but probably worth adding a user-visible note about when this becomes a feature.

**Comment by CryZe** [comment]
Yeah, this is completely impractical, as everyone is using completely different splits, so we don't even have any kind of consensus there. Not only when and how often they split, but split names as well. Also this would require a lot of moderation, as this would probably be auto submitted, so anyone miss-splitting (which apparently happens a lot), will accidently submit a wrong platinum time (that's what we used to call it). We didn't even have a global leaderboard one year ago (and some people still don't use speedrun.com), so not even World Records can be properly looked up, even though that doesn't have all of these issues with split names and co. So yeah, I don't see this happening in any good way at all.
